### DISCUSSION TITLE
please help

### DISCUSSION THREAD

**Discussion by ghost** [vosen/ZLUDA#187] [Q&A] [answered]
hi i tried to download the zluda zip extract it and run the zluda application only for windows to say it cant run the program or it opens then crashes immediately ive tried uninstalling it reinstalling it restarting pc and checking gpu drivers are up to date nothing helped anyone have any ideas ?

PC specs
amd ryzen 7 5800x3D 
amd radeon 7900xtx 
32GB ddr4 ram 
adata m.2 ssd
1000 watt psu 
asus tuf 570x-plus motherboard

**Comment by deuszex** [comment]
What are you trying to run with it? Just to make sure, zluda.exe is best used from the command line (example in the docs).

**Reply by ghost** [reply]
i tried it in cmd if thats what you mean by command line didn't do anything 
im not trying to run anything with it? im just trying to run zluda

**Reply by deuszex** [reply]
That's the problem, zluda is not an "application", it is and "extra layer" to run an application on.
Meaning if it has not other application targeted it will just shut off, because it has nothing to do.
Meaning it doesn't "crash" for you, it just shuts of as it should.

**Reply by ghost** [reply]
but if i cant go into the program how am i meant to tell it what application to work with ?

**Reply by deuszex** [reply]
[You use it from the command line (cmd) like in the example in the usage part of the documentation.](https://github.com/vosen/ZLUDA?tab=readme-ov-file#usage) <<< This is how you're meant to do it.

**Reply by ghost** [reply]
i did <ZLUDA_DIRECTORY>\zluda.exe -- <APPLICATION> <APPLICATION_ARGUMENTS> and it says was unexpected at this time ?

**Reply by deuszex** [reply]
Example:
You put zluda into blenders folder, then open the command line there and execute:
```
./zluda.exe -- ./blender.exe
```

Which for me in all actuality (don't paste this because your directory tree is different) goes like this on Windows:
```
./zluda.exe -- ../../../../Program\ Files/Blender\ Foundation/Blender\ 4.0/blender.exe
```
and this ^ opens blender for me with ZLUDA enabled on the CUDA tab under Cycles render devices.

**Reply by ghost** [reply]
![1tesrt](https://github.com/vosen/ZLUDA/assets/159965833/9e73aae1-418b-4032-96bb-4a0c0eb361fe)
![2tesrt](https://github.com/vosen/ZLUDA/assets/159965833/78e4915d-888c-4543-b729-ca668e37cc1b)
am i close to getting it to work ?
image 1 command you gave not working 
image 2 zluda folder in the blender folder

**Reply by deuszex** [reply]
Yeah, if you pulled the contents from the zluda folder out into the blender folder it would probably work, so zluda.exe and blender.exe are in the same folder.

Or you move into the zluda folder with the cmd then from there you designate the target, which should be (and this should work if I don't mess up something):
```
./zluda.exe -- ../blender.exe
```

You should read up on how cmd operates but the short version you should know here is
`./zluda.exe` -> means that zluda.exe is in the folder you're in with cmd
`../blender.exe` -> blender.exe is in the folder that contains the folder you're in with cmd.

```
Blender 4.0
- blender.exe
- zluda (the folder)
-- zluda.exe
```

**Reply by ghost** [reply]
![00000090](https://github.com/vosen/ZLUDA/assets/159965833/ee93834d-185b-4248-93b8-213af4dce052)
the command 
./zluda.exe -- ../blender.exe still not recognised

**Reply by deuszex** [reply]
Now they are in the same folder so remove a dot from befor blender.exe
`./zluda.exe -- ./blender.exe`

**Reply by ghost** [reply]
![helicoptergoobrrrr](https://github.com/vosen/ZLUDA/assets/159965833/5e6276a5-21fe-4659-8049-872d65cff786)

**Reply by deuszex** [reply]
Now you have too few dots before /blender. Just copy this as is without changes:
```
./zluda.exe -- ./blender.exe
```

**Reply by ghost** [reply]
im so confused you said get rid of the dot and now your saying dont change it ?
![pooooo](https://github.com/vosen/ZLUDA/assets/159965833/7bac0291-71af-4ccb-8f96-e2680f533fdd)

**Reply by deuszex** [reply, marked as answer]
I said either move the file, or write two dots. You moved and tried two dots, so I told you to remove one and wrote how it should look like, but you removed a dot from the one I sent and not the original, so you ended up without a dot, when needing one.


Anyway figured out why my help was ineffective, I'm using a unix style cmd replacement which takes commands like it does on linux. Tried from native windows cmd it works like this:
![image](https://github.com/vosen/ZLUDA/assets/30974986/09e3589b-b239-4c53-975b-eda9fc752c66)

with the command
```
zluda.exe -- blender.exe
```

So the command is pretty much the same you just leave out the ./ from the previous versions.

**Reply by ghost** [reply]
thanks it runs though its not behaving as i would expect can i ask you more questions ? do you know much about blender? and zluda ?
